engine/access/rest/blocks_test.go

```diff
@@ -0,0 +1,285 @@
+package rest
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	mocks "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/access/mock"
+	"github.com/onflow/flow-go/engine/access/rest/middleware"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type testVector struct {
+	description      string
+	request          *http.Request
+	expectedStatus   int
+	expectedResponse string
+}
+
+// TestGetBlocks tests the get blocks by ID and get blocks by heights API
+func TestGetBlocks(t *testing.T) {
+	backend := &mock.API{}
+
+	blkCnt := 10
+	blockIDs, heights, blocks, executionResults := generateMocks(backend, blkCnt)
+
+	singleBlockExpandedResponse := expectedBlockResponsesExpanded(blocks[:1], executionResults[:1], true)
```

Comment:
@smnzhu can't find your old comment now, but thanks for catching the fact that this was not a single block but multiple one. Fixed the index now to `:1` (instead of `:2`)